### ISSUE TITLE
SPIKE-2: Determine Best Option Of Object Recognition

### ISSUE THREAD

**Issue by cristian-aldea** [RGPosadas/LitterRally#9]
**Spike Description**

The trash recognition feature of our application is arguably the most technically uncertain part of the feature set. Although our proposal examined the possible options of libraries and/or approaches, the uncertainties and limitations of the approaches possible to us need to be explored.

* [ ] Document all viable approaches to in-browser (pref. real-time) object detection
* [ ] Create a proof-of-concept which can run in the browser and detect object from the user's camera

**Timebox**
2 weeks

**After the Timebox**
Don't forget to document all spike findings as comments within the spike issue.

**Comment by cristian-aldea** [comment]
List of classfications given by Google Vision API
https://stackoverflow.com/questions/38363182/is-there-a-full-list-of-potential-labels-that-googles-vision-api-will-return

**Comment by cristian-aldea** [comment]
Google doc where unrefined findings will be documented:
https://docs.google.com/document/d/1OJ0xyn6ViWZpZrgU8Ts2kAoVU0qea3qSE-rardicFEg/edit

**Comment by ys-lin** [comment]
proof of concept: https://github.com/ys-lin/recyclable-material-detection

**Comment by cristian-aldea** [comment]
I adjusted the requirements for the spikes, since some requirements were no longer relevant for #33 

The spike will be closed today, we feel that we've spent more than enough time researching this topic. We've come up with a few viable options, some of which we were able to make functional proof-of-concept implementations.

See https://docs.google.com/document/d/1OJ0xyn6ViWZpZrgU8Ts2kAoVU0qea3qSE-rardicFEg/edit for details

The approach that is currently the most viable is training a YOLO model with darknet, converting that model into a format that can be run by tensorflow.js, and then using the https://github.com/shaqian/tfjs-yolo library to run the model.

Other approaches could also work, but they either had too many limitations or were too difficult or long to implement into a working POC.
